### PR TITLE
go mod: update the parser version

### PR DESCRIPTION
**Pull request by winkyao**
### What problem does this PR solve? 

It solved the mistake imported by #8037, I merge the TiDB pr before the https://github.com/pingcap/parser/pull/14 merged.

Maybe we should find a way to avoid this thing happen again.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8632)

### CODE REVIEW

**Comment by zz-jason** [comment]
> @zz-jason But we need the replace to verify the merging parser pr is work.

You can add it as the last step of `check`:
```makefile
 57 dev: checklist test check

 ...
 67 check: check-setup fmt lint vet #thirdparser
```

This check progress will started once all the unit tests are passed

Or you can add this check in `circle.yml`, like `go mod tidy`, see https://github.com/pingcap/tidb/pull/8402 for more detail.

**Comment by dbjoa** [comment]
@winkyao ``master`` branch has been broken. Could you recover your local branch  ``github.com/winkyao/parser v0.0.0-20181207053352-37ef416fe2a0`` ?
```
CGO_ENABLED=0 GO111MODULE=on go build   -ldflags '-X "github.com/pingcap/parser/mysql.TiDBReleaseVersion=v2.1.0-rc.3-300-g3feb22b" -X "github.com/pingcap/tidb/util/printer.TiDBBuildTS=2018-12-10 06:37:20" -X "github.com/pingcap/tidb/util/printer.TiDBGitHash=3feb22b291229ac5bcef39bcbc8a5428a92ed6bc" -X "github.com/pingcap/tidb/util/printer.TiDBGitBranch=fix-the-issue-8615" -X "github.com/pingcap/tidb/util/printer.GoVersion=go version go1.11 linux/amd64" ' -o bin/tidb-server tidb-server/main.go
go: finding github.com/winkyao/parser v0.0.0-20181207053352-37ef416fe2a0
go: github.com/winkyao/parser@v0.0.0-20181207053352-37ef416fe2a0: unknown revision 37ef416fe2a0
go: error loading module requirements
Makefile:161: recipe for target 'server' failed
make: *** [server] Error 1
```

**Comment by winkyao** [comment]
@dbjoa Fine, I push the local branch to the github. Please try to rerun the unit tests.

**Comment by dbjoa** [comment]
OK, recovered!

**Comment by winkyao** [comment]
@zz-jason Well, I'll figure it out.

File: planner/core/logical_plan_test.go [hunk 1]
```diff
@@ -993,7 +993,7 @@ func (s *testPlanSuite) TestColumnPruning(c *C) {
 		},
 		//issue 7833
 		{
-			sql: "drop view if exists v",
```
Comment (XuHuaiyu) [inline planner/core/logical_plan_test.go]: We can just remove it.
We'll add other test cases for drop view later.

**Pull request merged**